packages/rollup/src/worker.js

```diff
@@ -0,0 +1,217 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * A bazel worker process for running rollup.
+ *
+ * Arguments:
+ *  --input.* value
+ *      Param and value to put into rollup inputOptions
+ *
+ *  --output.* value
+ *      Param and value to put into rollup outputOptions
+ *
+ *  --entry path/to/entry
+ *      The entry file, converted to rollup "input"
+ *
+ *  --entries x=path/to/x y=path/to/y
+ *      Multiple entry files, converted to rollup "input"
+ *
+ *  --config x
+ *      Custom rollup config file. Some options overridden by bazel
+ */
+
+const path = require('path');
+const rollup = require('rollup');
+let worker;
+try {
+  worker = require('@bazel/worker')
+} catch {
+  // TODO: rely on the linker to link the first-party package
+  const helper = process.env['BAZEL_NODE_RUNFILES_HELPER'];
+  if (!helper) throw new Error('No runfiles helper and no @bazel/worker npm package');
+  const runfiles = require(helper);
+  const workerRequire =
+      runfiles.resolve('build_bazel_rules_nodejs/packages/worker/npm_package/index.js');
+  if (!workerRequire)
+    throw new Error(
+        `build_bazel_rules_nodejs/packages/worker/npm_package/index.js missing in runfiles ${
+            JSON.stringify(runfiles.manifest)}, ${runfiles.dir}`);
+  worker = require(workerRequire);
+}
+
+const MNEMONIC = 'Rollup';
+
+const INPUT_ARG_PREFIX = '--input.';
+const OUTPUT_ARG_PREFIX = '--output.';
+
+// Parse the worker CLI args from rollup_bundle.bzl into rollup input/outputOptions
+async function parseCLIArgs(args) {
+  let entries = null;
+  let inputOptions = {};
+  let outputOptions = {};
+  let configFile = null;
+
+  for (let i = 0; i < args.length; i++) {
+    const arg = args[i];
+    if (arg.startsWith(INPUT_ARG_PREFIX)) {
+      inputOptions[arg.slice(INPUT_ARG_PREFIX.length)] = args[++i];
+    } else if (arg.startsWith(OUTPUT_ARG_PREFIX)) {
+      outputOptions[arg.slice(OUTPUT_ARG_PREFIX.length)] = args[++i];
+    } else if (arg === '--config') {
+      configFile = args[++i];
+    } else if (arg === '--entry') {
+      if (entries) {
+        throw new Error(`${MNEMONIC}: only one of --entry or --entries`);
+      }
+
+      entries = args[++i];
+    } else if (arg === '--entries') {
+      if (entries) {
+        throw new Error(`${MNEMONIC}: only one of --entry or --entries`);
+      }
+
+      entries = {};
+      do {
+        const [entryName, entryPath] = args[++i].split('=', 2);
+        entries[entryName] = entryPath;
+      } while (i < args.length && !args[i + 1].startsWith('--'));
+    } else {
+      throw new Error(`${MNEMONIC}: unknown worker param: ${arg}`);
+    }
+  }
+
+  // Additional options passed via config file
+  // TODO: only on first run or if bazel says it changed
```

Comment:
This is probably worth doing at some point. Doesn't have to be in this initial PR, but only running `loadConfigFile()` when the config (and its deps?) changes would avoid a rollup run per worker loop...